lib/utilities/collectCorsHeaders.js

```diff
@@ -0,0 +1,34 @@
+import { findCorsRule,
+generateCorsResHeaders } from '../api/apiUtils/object/corsResponse.js';
+
+/**
+ * collectCorsHeaders - gather any relevant CORS headers
+ * @param {object} request - http request object
+ * @param {BucketInfo} bucket - instance of BucketInfo class
+ * @return {object} - object containing CORS headers
+ */
+export default function collectCorsHeaders(request, bucket) {
+    const origin = request.headers.origin;
+    const method = request.method;
+
+    // NOTE: Because collecting CORS headers requires making a call to
+    // metadata to retrieve the bucket's CORS configuration, we opt not to
+    // return the CORS headers if the request encounters an error before
+    // the api method retrieves the bucket from metadata (an example
+    // being if a request is not properly authenticated). This is a slight
+    // deviation from AWS compatibility, but rebuffs additional backend calls,
+    // anticipating that the preflight OPTIONS route will serve most client
+    // needs regarding CORS.
+    if (!origin || !bucket) {
+        return null;
```

Comment:
I have updated it but it seems to have turned up an undetected issue:
in some cases, I was returning `responseMetaHeaders` including `corsHeaders` in case of error when we only want to return `corsHeaders`. This caused one of the legacy tests for objectGet to fail. I'm working on fixing it.